drivers/i2c/i2c.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <delay.h>
+#include <dm.h>
+#include <i2c.h>
+#include <mmio.h>
+#include <stddef.h>
+#include <util.h>
+
+#define I2C_STAT_REG 0x10
+
+#define I2C_TIMEOUT  10 /*< Timeout interval (in cycles). */
```

Comment:
This timeout needs to be based on the frequency of the I²C bus (100kHz). So the total timeout is 100μs = 10μs/clock * 10 clocks (8 data bits + 1 ack + 1 margin for error).